quesma/quesma/search.go

```
@@ -164,6 +166,106 @@ func (q *QueryRunner) HandleCount(ctx context.Context, indexPattern string) (int
 	}
 }
 
+func (q *QueryRunner) HandleMultiSearch(ctx context.Context, defaultIndexName string, body types.NDJSON) ([]byte, error) {
+
+	type msearchQuery struct {
+		indexName string
+		query     types.JSON
+	}
+
+	var queries []msearchQuery
+
+	var currentQuery *msearchQuery
+
+	for _, line := range body {
+
+		if currentQuery == nil {
+			currentQuery = &msearchQuery{}
+
+			if v, ok := line["index"].(string); ok {
+				currentQuery.indexName = v
+			} else {
+				currentQuery.indexName = defaultIndexName
+			}
+		} else {
+
+			q := types.JSON{}
+			q["query"] = line["query"]
```

Should not we checked whether `query` exists?

---

It's required. We should raise an error if not provided.

---

ok, but do we do that? I don't see it, maybe at different layer?

---

I've just added it.